lib/Doctrine/DBAL/Schema/DB2SchemaManager.php

```diff
@@ -204,4 +205,18 @@ protected function _getPortableViewDefinition($view)
 
         return new View($view['name'], $sql);
     }
+
+    public function listTableDetails($tableName)
+    {
+        $table = parent::listTableDetails($tableName);
+
+        /** @var DB2Platform $platform */
```

Comment:
There seems to be some repetition of the code like this: most platforms override `listTableDetails()`, call the parent and add the comment. Probably, this logic should be part of the parent method itself (most of the platforms support comments).

Reply:
True, but there is a catch.

For most DBMS we want to fetch the comments. For MySQL, we fetch the comments + a bunch of metadata.

What we could do is design a method that returns an array of "options" to be applied to the table. Something like:

```php
    public function listTableDetails($tableName)
    {
        // ....
        $options      = $this->getTableOptions($tableName);
        foreach ($options as $key => $value) {
            $table->addOption($key, $value);
        }
        return $table;
    }
```

Now, for each `XXXSchemaManager`, we can provide a protected `getTableOptions` method that is specific to the schema manager. Would that be ok?